pravega/pravega-operator
# Issue-468: Run pravega segmentstore sts with runAsUser = 0

Signed-off-by: prabhaker24 <prabhaker.saxena@dell.com>

### Change log description
Pravega with ECS as LTS is failing to deploy in Openshift environment, as container needs to inject cert bundle into trust store, but the security context selected by openshift is preventing it.
To solve the above problem we have to set runAsUser = 0 for  pravega segmentstore sts.

### Purpose of the change
Fixes #468 

### What the code does
These changes allow the segmentstore pod's to run as root user. For enabling this user has to give the following things in the manifest file while deploying it manually

```
 pravega:
   segmenStoreSecurityContext: 
    runAsUser: 0  
```
If he is using charts he needs to uncomment the line in the values.yaml in the pravega charts:-

```
segmentStoreSecurityContext: 
  runAsUser: 0

```
For controller he needs to set like this:-

```
controllerSecurityContext:
   runAsUser: 0
```

### How to verify it
Deploy the pravega cluster with **runAsUser: 0** and check the segment store sts and controller deployment post-deployment  it should have the following in it's described output

```
      securityContext:
        runAsUser: 0
```
